src/drivers/serac.cpp

```diff
@@ -1,206 +1,206 @@
-    SLIC_ERROR_ROOT(rank, err_msg);
-    serac::exitGracefully();
-  }
-
-  auto config_msg = app.config_to_str(true, true);
-  SLIC_INFO_ROOT(rank, config_msg);
-
-  auto mesh = serac::buildParallelMesh(mesh_file, ser_ref_levels, par_ref_levels);
-
-  int dim = mesh->Dimension();
-
-  // Define the solid solver object
-  serac::NonlinearSolidSolver solid_solver(order, mesh);
-
-  // Project the initial and reference configuration functions onto the
-  // appropriate grid functions
-  mfem::VectorFunctionCoefficient defo_coef(dim, serac::initialDeformation);
-
-  mfem::Vector velo(dim);
-  velo = 0.0;
-
-  mfem::VectorConstantCoefficient velo_coef(velo);
-
-  // initialize x_cur, boundary condition, deformation, and
-  // incremental nodal displacment grid functions by projection the
-  // VectorFunctionCoefficient function onto them
-  solid_solver.setDisplacement(defo_coef);
-  solid_solver.setVelocity(velo_coef);
-
-  std::set<int> ess_bdr = {1};
-
-  // define the displacement vector
-  mfem::Vector disp(dim);
-  disp           = 0.0;
-  auto disp_coef = std::make_shared<mfem::VectorConstantCoefficient>(disp);
-
-  std::set<int> trac_bdr = {2};
-
-  // define the traction vector
-  mfem::Vector traction(dim);
-  traction(0) = tx;
-  traction(1) = ty;
-  if (dim == 3) {
-    traction(2) = tz;
-  }
-
-  auto traction_coef = std::make_shared<serac::VectorScaledConstantCoefficient>(traction);
-
-  // Set the boundary condition information
-  solid_solver.setDisplacementBCs(ess_bdr, disp_coef);
-  solid_solver.setTractionBCs(trac_bdr, traction_coef);
-
-  // Set the material parameters
-  solid_solver.setHyperelasticMaterialParameters(mu, K);
-
-  // Set the linear solver parameters
-  if (gmres_solver == true) {
-    lin_params.prec       = serac::Preconditioner::BoomerAMG;
-    lin_params.lin_solver = serac::LinearSolver::GMRES;
-  } else {
-    lin_params.prec       = serac::Preconditioner::Jacobi;
-    lin_params.lin_solver = serac::LinearSolver::MINRES;
-  }
-  solid_solver.setSolverParameters(lin_params, nonlin_params);
-
-  // Set the time step method
-  solid_solver.setTimestepper(serac::TimestepMethod::QuasiStatic);
-
-  // Complete the solver setup
-  solid_solver.completeSetup();
-
-  // initialize/set the time
-  double t = 0.0;
-
-  bool last_step = false;
-
-  solid_solver.initializeOutput(serac::OutputType::VisIt, "serac");
-
-  // enter the time step loop. This was modeled after example 10p.
-  for (int ti = 1; !last_step; ti++) {
-    double dt_real = std::min(dt, t_final - t);
-    // compute current time
-    t = t + dt_real;
-
-    if (rank == 0) {
-      std::cout << "step " << ti << ", t = " << t << std::endl;
-    }
-
-    traction_coef->SetScale(t);
-
-    // Solve the Newton system
-    solid_solver.advanceTimestep(dt_real);
-
-    solid_solver.outputState();
-
-    last_step = (t >= t_final - 1e-8 * dt);
-  }
-
-  serac::exitGracefully();
-}
+// Copyright (c) 2019-2020, Lawrence Livermore National Security, LLC and

```

Comment:
We are losing a lot of history/blame utility due to whatever you changed on these files that you are touching every line.

Reply:
can you help me figure out how to fix the issue?

Reply:
It may be a switch from LF->CRLF

Reply:
It looks like it was done to 3? files, if it was the last commit, you can do the following:

```
// undo last commit locally but keep changes
git reset --soft HEAD~1
git checkout -- path/to/files
// commit changes you actually wanted
```

If it is mixed into your history, its often easier to just start from scratch and copy your work into another branch and/or cherry pick SHAs.

I can help you with either after I eat lunch.

Reply:
Though to be honest we are early in the project and these files will be drastically changed soon.  So maybe just fix the current probably line ending problem and try not to do it in the future. @jamiebramwell thoughts?

Reply:
I think that sounds good. As long as the line endings are OK, I think a muddled history is fine.

Reply:
at least undo the last commit which contained 3 of the offending files.

Reply:
After some googling, it looks like this issue might be avoidable in the future by creating a .gitattributes file to specify line ending conventions. I don't know enough about that approach to have an opinion, does anyone else have thoughts on adding a .gitattributes file for serac?

Reply:
This seems like a good idea. https://docs.github.com/en/github/using-git/configuring-git-to-handle-line-endings